lib/SILOptimizer/Utils/Generics.cpp

```diff
@@ -3119,6 +3119,12 @@ void swift::trySpecializeApplyOfGeneric(
                             << SpecializedF->getLoweredFunctionType() << "\n");
     NewFunctions.push_back(SpecializedF.getFunction());
   }
+  if (replacePartialApplyWithoutReabstraction &&
+      SpecializedF.getFunction()->isExternalDeclaration()) {
+    // Cannot create a tunk without having the body of the function.
```

Comment:
If you create a main PR to fix the "tunk" typo, you could also explain that thunk creation needs to access the callee function's SIL arguments.